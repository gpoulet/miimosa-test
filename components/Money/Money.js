import LinearProgress from '@material-ui/core/LinearProgress'
import Typography from '@material-ui/core/Typography'
import { number, string } from 'prop-types'
import { getColorFromStatus, getPercentValue, getProgressValue } from './Money.util'

function Money({ collected, target, status }) {
  let value = getProgressValue(collected, target)
  let percent = getPercentValue(collected, target)
  return (
    <>
      <Typography variant="body2" color="textSecondary" component="p">
        {collected + "€ collectés sur " + target + "€ (" + percent + "%)"}
      </Typography>
      <LinearProgress variant="determinate" value={value} color={getColorFromStatus(status)}/>
    </>
  )
}

Money.propTypes = {
  collected: number.isRequired,
  target: number.isRequired,
  status: string.isRequired
}

export default Money