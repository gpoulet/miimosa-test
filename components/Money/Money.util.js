export const getProgressValue = (collected, target) => {
  const value = Number(collected * 100 / target)
  if (value > 100) {
    return 100
  } else {
    return value;
  }
}

export const getPercentValue = (collected, target) => {
  return Math.floor(Number(collected * 100 / target))
}

export const getColorFromStatus = status => {
  switch (status) {
    case "success": return "primary"
    case "in_progress": return "secondary"
    default : return "secondary"
  }
}