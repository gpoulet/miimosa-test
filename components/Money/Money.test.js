import React from 'react';
import renderer from 'react-test-renderer';
import Money from './Money';

it('Money renders correctly', () => {
  const tree = renderer
    .create(<Money collected={15000} target={20000} status="in_progess" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});