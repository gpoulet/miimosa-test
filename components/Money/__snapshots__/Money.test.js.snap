// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Money renders correctly 1`] = `
Array [
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
  >
    15000€ collectés sur 20000€ (75%)
  </p>,
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={75}
    className="MuiLinearProgress-root MuiLinearProgress-colorSecondary MuiLinearProgress-determinate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Determinate"
      style={
        Object {
          "transform": "translateX(-25%)",
        }
      }
    />
  </div>,
]
`;
